### ISSUE TITLE
Global install doesn't use %LocalAppData\Yarn\.bin

### ISSUE THREAD

**Issue by onemen** [yarnpkg/yarn#1226]
**Do you want to request a _feature_ or report a _bug_?**
Bug

**What is the current behavior?**
The Windows installer add %LocalAppData\Yarn.bin to the PATH (#1129).
After using `yarn global add [packages ...]` i did not found %LocalAppData\Yarn\.bin, instead the bin is in %LocalAppData\Yarn\config\global\node_modules\.bin

Additionally `yarn global bin` is C:\Program Files\nodejs, so cmd are link from C:\Program Files\nodejs to %LocalAppData\Yarn\config\global\node_modules\.bin
we end up with three-way link:
nodejs ---> node_modules\.bin ---> package\.bin

**If the current behavior is a bug, please provide the steps to reproduce.**
`yarn global add [packages ...]`

**What is the expected behavior?**
1. global install should cmd-shims all the bins into %LocalAppData\Yarn\.bin instead of %LocalAppData\Yarn\config\global\node_modules\.bin
2. `yarn global bin` should point to %LocalAppData\Yarn\.bin
3. Remove the extra step of linking to C:\Program Files\nodejs 

**Please mention your node.js, yarn and operating system version.**
Windows 10 Pro x64 v1607
node v6.9.0
yarn v0.16.0

**Comment by Loyalsoldier** [comment]
I came across the same problem. And this is the main reason I do not use Yarn at the first time.

**Comment by Daniel15** [comment]
This was fixed in 0.24.x

**Comment by Loyalsoldier** [comment]
`OS:` Windows 10 Creators Update
`node version:` v6.10.3 (with npm v3.10.10)
`yarn version:` v0.24.5
`Terminal:` CMD and Cmder

If there is a "prefix" key-value in npm config file, yarn's globally added package would not be found because the "prefix" value(the path set for npm globally added packages) would be add before the yarn global bin path, and CMD or Cmder could not find that non-existed path.

**npm user config:**

```
depth=0
loglevel=http
registry=https://registry.npm.taobao.org/
prefix=D:\dev\software\nodejs\node_global
```

After globally added `gulp` with yarn and gulp's command could not be found with the following error using Cmder (BTW, CMD would not throw this error, it was just without output):

```
> yarn global add gulp
yarn global v0.24.5
warning No license field
[1/4] Resolving packages...
warning gulp > vinyl-fs > glob-stream > minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning gulp > vinyl-fs > glob-watcher > gaze > globule > minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning gulp > vinyl-fs > glob-watcher > gaze > globule > glob > graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
[2/4] Fetching packages...
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
█████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
███████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
██████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
██████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
██████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
█████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
████████████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
█████████████████████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
████████████████████████████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
██████████████████████████████████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░
██████████████████████████████████████████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░
█████████████████████████████████████████████████████████████████████████████████████████████████████████████░░░░░░░░░░
█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████░░
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
warning fsevents@1.1.1: The platform "win32" is incompatible with this module.
info "fsevents@1.1.1" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
█████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
███████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
███████████████████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
█████████████████████████████████████████████████████████████████████████████████████████████████████████████████░░░░░░
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
██████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
█████████████████████████████████████████████████████████████████████████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████
[4/4] Building fresh packages...
success Installed "gulp@3.9.1" with binaries:
      - gulp
warning No license field
Done in 11.42s.


> gulp -v
/bin/sh: /d/dev/software/nodejs/node_global/C:/Users/myUserName/AppData/Local/Yarn/config/global/node_modules/.bin/gulp: No such file or directory
```

The path `/d/dev/software/nodejs/node_global/` in the error is the global packages path I set for npm. And `C:/Users/myUserName/AppData/Local/Yarn/config/global/node_modules/.bin/gulp` is the global bin path for Yarn.
